packages/dev/core/src/Lights/areaLight.ts

```diff
@@ -0,0 +1,93 @@
+import type { Vector3 } from "core/Maths/math.vector";
+import { Light } from "core/Lights/light";
+import type { Effect } from "core/Materials/effect";
+import { DefaultAreaLightLTCProvider } from "core/Lights/LTC/ltcTextureTool";
+import type { IAreaLightLTCProvider, ILTCTextures } from "core/Lights/LTC/ltcTextureTool";
+import type { Scene } from "core/scene";
+import { Logger } from "core/Misc/logger";
+
+declare module "../scene" {
+    export interface Scene {
+        /**
+         * Object capable of providing LTC textures for Area Lights.
+         */
+        areaLightLTCProvider?: IAreaLightLTCProvider;
+
+        /**
+         * @internal
+         */
+        _ltcTexturesPromise?: Promise<ILTCTextures>;
+
+        /**
+         * @internal
+         */
+        _ltcTextures?: ILTCTextures;
+    }
+}
+
+/**
+ * Abstract Area Light class that servers as parent for all Area Lights implementations.
+ * The light is emitted from the area in the -Z direction.
+ */
+export abstract class AreaLight extends Light {
+    /**
+     * Area Light position.
+     */
+    public position: Vector3;
+
+    /**
+     * Creates a area light object.
+     * Documentation : https://doc.babylonjs.com/features/featuresDeepDive/lights/lights_introduction
+     * @param name The friendly name of the light
+     * @param position The position of the area light.
+     * @param scene The scene the light belongs to
+     */
+    constructor(name: string, position: Vector3, scene?: Scene) {
+        super(name, scene);
+        this.position = position;
+
+        if (!this._scene._ltcTextures) {
+            this._scene.areaLightLTCProvider ||= new DefaultAreaLightLTCProvider(this._scene, "https://assets.babylonjs.com/areaLights/areaLightsLTC.bin");

```

Comment:
After this PR #16089 is merged and available in master you can use the same paradigm to allow overriding this URL (i.e. use the Tools.GetAssetsUrl function). Would also be good if you moved the asset to the /core/ directory in the Assets repository to stay consistent and make the asset override procedure work correctly.